src/vs/workbench/contrib/debug/browser/breakpointEditorContribution.ts

```diff
@@ -235,39 +236,54 @@ export class BreakpointEditorContribution implements IBreakpointEditorContributi
 
 				if (breakpoints.length) {
+					const isShiftPressed = e.event.shiftKey;
+
 					// Show the dialog if there is a potential condition to be accidently lost.
 					// Do not show dialog on linux due to electron issue freezing the mouse #50026
 					if (!env.isLinux && breakpoints.some(bp => !!bp.condition || !!bp.logMessage || !!bp.hitCondition)) {
 						const logPoint = breakpoints.every(bp => !!bp.logMessage);
 						const breakpointType = logPoint ? nls.localize('logPoint', "Logpoint") : nls.localize('breakpoint', "Breakpoint");
-						const disable = breakpoints.some(bp => bp.enabled);
+						const enabled = breakpoints.some(bp => bp.enabled);
+						const isDisableAction = enabled && isShiftPressed;
 
-						const enabling = nls.localize('breakpointHasConditionDisabled',
+						const disabledBreakpointDialogMessage = nls.localize(
+							'breakpointHasConditionDisabled',
 							"This {0} has a {1} that will get lost on remove. Consider enabling the {0} instead.",
 							breakpointType.toLowerCase(),
 							logPoint ? nls.localize('message', "message") : nls.localize('condition', "condition")
 						);
-						const disabling = nls.localize('breakpointHasConditionEnabled',
+						const enabledBreakpointDialogMessage = nls.localize(
+							'breakpointHasConditionEnabled',
 							"This {0} has a {1} that will get lost on remove. Consider disabling the {0} instead.",
 							breakpointType.toLowerCase(),
 							logPoint ? nls.localize('message', "message") : nls.localize('condition', "condition")
 						);
 
-						const { choice } = await this.dialogService.show(severity.Info, disable ? disabling : enabling, [
-							nls.localize('removeLogPoint', "Remove {0}", breakpointType),
-							nls.localize('disableLogPoint', "{0} {1}", disable ? nls.localize('disable', "Disable") : nls.localize('enable', "Enable"), breakpointType),
-							nls.localize('cancel', "Cancel")
-						], { cancelId: 2 });
+						const dialogActionChoicePromise = isDisableAction
+							? Promise.resolve(TOGGLE_BREAKPOINT_BUTTON_INDEX)
+							: this.dialogService.show(
+								severity.Info,
+								enabled ? enabledBreakpointDialogMessage : disabledBreakpointDialogMessage,
+								[
+									nls.localize('removeLogPoint', "Remove {0}", breakpointType),
+									nls.localize('disableLogPoint', "{0} {1}", enabled ? nls.localize('disable', "Disable") : nls.localize('enable', "Enable"), breakpointType),
+									nls.localize('cancel', "Cancel")
+								],
+								{ cancelId: 2 },
+							).then(({ choice }) => choice);
+
+						const choice = await dialogActionChoicePromise;
 
 						if (choice === 0) {
 							breakpoints.forEach(bp => this.debugService.removeBreakpoints(bp.getId()));
 						}
-						if (choice === 1) {
-							breakpoints.forEach(bp => this.debugService.enableOrDisableBreakpoints(!disable, bp));
+						if (choice === TOGGLE_BREAKPOINT_BUTTON_INDEX) {
```

Comment:
This isn't really more clear, because the order is determined just a few lines above, so I think hardcoding the 1 is more readable